src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -166,6 +166,9 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, Action<Ht
         // We consider calls to `UseHttps` to be a clear expression of user intent to pull in HTTPS configuration support
         listenOptions.KestrelServerOptions.EnableHttpsConfiguration();
 
+        // If there's a configuration, load it so that the results will be available to ApplyDefaultCertificate
+        listenOptions.KestrelServerOptions.ConfigurationLoader?.LoadInternal();
```

Comment:
Does this load the dev cert too? Are there additional console logs for that?

Reply:
Assuming you mean the one from the cert store (and not the magic docker one), I don't believe so, but I can double check.  And I'm not sure what you mean by "additional console logs"?

Reply:
In a prior release we'd changed the dev cert to delay-load from the cert store because if it loaded earlier then even apps with their own cert would get logs about the dev cert being loaded, and it was a startup time perf hit.

Reply:
This is not changing when any cert gets loaded. I might help find the default cert first and elide the cert store check, but the cert store would still be loaded later in `ApplyDefaultCertificate` as usual otherwise.

Reply:
I keep getting mixed up because of the number of moving parts (not to mention the number of iterations this change has been through), but I think it _can_ cause the cert to be loaded earlier.  However, unlike (what I understand of) the previous change, it will only do that in cases where it was going to be loaded regardless (except for the weird corner case where you add a cert to your config between `UseHttps` and `Bind`).

Reply:
Does that address your concerns, @Tratcher?